### PR TITLE
[3722] Write compose performance guidelines

### PR DESCRIPTION
**Pull request by MarinTolic**
### 🎯 Goal

Satisfies the first acceptance criteria of #3722 

### 🛠 Implementation details

Wrote a guide that touched upon the most common mistakes found in our codebase.

### 🧪 Testing

1. Read the guideline
2. Suggestions are welcome
3. Please comment on how informative this was, if it will help you write more performant code going forward. etc.

#### General
- [ ] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [ ] Assigned a person / code owner group (required)
- [ ] Thread with the PR link started in a respective Slack channel (#android-chat-core or #android-chat-ui) (required)
- [ ] PR targets the `develop` branch
- [ ] PR is linked to the GitHub issue it resolves

#### Code & documentation
- [x] Changelog is updated with client-facing changes

### ☑️Reviewer Checklist
- [ ] Read the guide and verify it makes sense and is informative

### 🎉 GIF

gif

### CODE REVIEW

**Review by TGazica — APPROVED** [review]
Overall looks and reads well, has good information on what to keep an eye out for.

File: stream-chat-android-compose/GUIDELINES.md [hunk 1]
```diff
@@ -92,7 +127,8 @@ When writing code comments, keep the following in mind:
 
 ### Naming Components
 
-When choosing a name for your component, try to avoid the naming conventions from Android view-based system.
+When choosing a name for your component, try to avoid the naming conventions from Android view-based
```
Comment (bychkovdmitry) [inline stream-chat-android-compose/GUIDELINES.md:130]: So, new lines are just ignored?

**Review by bychkovdmitry — APPROVED** [review]
Tremendous work, looks good!

**Review by filbabic — CHANGES_REQUESTED** [review]
I'll go through an edit pass here and do some cleanup - there are typos lying around (non-issue), but I think the structure could be improved as this is a huge guide that could probably live on its own.

Also - this should be in docusaurus too, so that our users can benefit from such guidelines.

File: stream-chat-android-compose/GUIDELINES.md [hunk 2]
```diff
@@ -5,27 +5,49 @@
...
-  * [Customization Options](#customization-options)
-  * [Slot APIs](#slot-apis)
+    * [Customization Options](#customization-options)
+    * [Slot APIs](#slot-apis)
 * [Drawable colors](#drawable-colors)
-* [Component Previews](#component-previews)  
+* [Component Previews](#component-previews)
+* [Performance Guidelines](#performance-guidelines)
+    * [Write Stable Classes](#write-stable-classes)
+    * [Real World Implications](#real-world-implications)
```
Comment (skydoves) [inline stream-chat-android-compose/GUIDELINES.md:21]: We published a new article following our performance guidelines.
Can we also mention this post somewhere? [6 Jetpack Compose Guidelines to Optimize Your App Performance](https://getstream.io/blog/jetpack-compose-guidelines/)

Comment (filbabic) [reply stream-chat-android-compose/GUIDELINES.md:21]: We'll work on condensing this info and publishing a shorter set of guidelines, by pointing to the post first, for people to learn the basics.

**Pull request closed without merging**